core/iwasm/common/wasm_runtime_common.c

```diff
@@ -7071,3 +7072,158 @@ wasm_runtime_detect_native_stack_overflow_size(WASMExecEnv *exec_env,
     }
     return true;
 }
+
+#if WASM_ENABLE_INTERP != 0
+static void
+destroy_all_wasm_sections(wasm_section_list_t sections)
+{
+    wasm_section_t *cur = sections;
+    while (cur) {
+        wasm_section_t *next = cur->next;
+        if (cur->section_body != NULL)
+            wasm_runtime_free(cur->section_body);
+        wasm_runtime_free(cur);
+        cur = next;
+    }
+}
+
+static void
+destroy_part_wasm_sections(wasm_section_list_t *p_sections,
+                           uint8 *section_types, int section_cnt)
+{
+    int i;
+    for (i = 0; i < section_cnt; i++) {
+        uint8 section_type = section_types[i];
+        wasm_section_t *cur = *p_sections, *prev = NULL;
+
+        while (cur) {
+            wasm_section_t *next = cur->next;
+            if (cur->section_type == section_type) {
+                if (prev)
+                    prev->next = next;
+                else
+                    *p_sections = next;
+
+                if (cur->section_body != NULL)
+                    wasm_runtime_free(cur->section_body);
```

Comment:
The section_body is just referring to the wasm file buf, so if we didn't add a new argument for create_sections_wasm and didn't allocate memory for section_body, here we cannot free it.